include/lsst/jointcal/Chi2.h

```diff
@@ -28,20 +44,70 @@ struct Chi2 {
         return s.str();
     }
 
-    // Addentry has a third argument in order to make it compatible with an
-    // other stat accumulator.
-    template <typename T>
-    void addEntry(double inc, unsigned dof, T) {
+    // Addentry has an ignored third argument in order to make it compatible with Chi2List.
+    void addEntry(double inc, unsigned dof, std::shared_ptr<BaseStar>) {
         chi2 += inc;
         ndof += dof;
     }
 
-    void operator+=(const Chi2& right) {
-        chi2 += right.chi2;
-        ndof += right.ndof;
+    void operator+=(const Chi2Statistic& rhs) {
+        chi2 += rhs.chi2;
+        ndof += rhs.ndof;
+    }
+};
+
+/*
+ * A class to accumulate chi2 contributions together with pointers to the contributors.
+ *
+ * This structure lets one compute the chi2 statistics (average and variance) and directly point back
+ * to the bad guys without relooping.
+ * The Chi2Star routine makes it compatible with AstrometryFit's
+ * accumulateStatImage and accumulateStatImageList.
+ */
+struct Chi2Star {
+    double chi2;
+    std::shared_ptr<BaseStar> star;
+
+    Chi2Star(double chi2, std::shared_ptr<BaseStar> star) : chi2(chi2), star(std::move(star)) {}
+    // for sorting
+    bool operator<(const Chi2Star& rhs) const { return (chi2 < rhs.chi2); }
+
+    friend std::ostream& operator<<(std::ostream& s, const Chi2Star& chi2Star) {
+        s << "chi2: " << chi2Star.chi2 << " star: " << *(chi2Star.star) << std::endl;
+        return s;
+    }
+};
+
+/// Structure to accumulate the chi2 contributions per each star (to help find outliers).
+class Chi2List : public Chi2Accumulator, public std::vector<Chi2Star> {
+public:
+    void addEntry(double chi2, unsigned ndof, std::shared_ptr<BaseStar> star) {
+        this->push_back(Chi2Star(chi2, std::move(star)));
+    }
+
+    /// Compute the average and std-deviation of these chisq values.
+    std::pair<double, double> computeAverageAndSigma() {
+        double sum = 0;
+        double sum2 = 0;
+        for (auto i : *this) {
+            sum += i.chi2;
+            sum2 += sqr(i.chi2);
+        }
+        double average = sum / this->size();
+        double sigma = sqrt(sum2 / this->size() - sqr(average));
+        return std::make_pair(average, sigma);
+    }
```

Comment:
There isn't a .cc file for Chi2: should I create one just for this method? I don't expect it will need any more methods, except the aforementioned `std::vector` passthroughs which are going to be short.